linux/install-dkms

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+#
+# Usage: ./install-dkms [640 480]
+
+set -e
+REPO_PATH="$(/usr/bin/dirname $([ -L $0 ] && /bin/readlink -f $0 || echo $0))"
+source "${REPO_PATH}/install_shell_func_and_conf"
+
+
+check_params "$1" "$2"
+
+# Installing droidcam binaries. Requires make to have succeeded.
+if [ -f "${REPO_PATH}/droidcam" ]; then 
+    echo "Installing droidcam in /usr/bin" 
+    cp "${REPO_PATH}/droidcam" /usr/bin/
+else
+    echo "droidcam was not previously built. Installation skip."
+fi
+if [ -f "${REPO_PATH}/droidcam-cli" ]; then 
+    echo "Installing droidcam-cli in /usr/bin" 
+    cp "${REPO_PATH}/droidcam-cli" /usr/bin/
+else
+    echo "droidcam-cli was not previously built. Installation skip."
+fi
+
+# Now setting up the module as a DKMS module
+
+# Creating link with correct name in /usr/src to the v4l2loopback directory
+cd "${REPO_PATH}/${V4L2_LOOPBACK_DIR}"
+if [ -d "/usr/src/${V4L2_LOOPBACK_DC}-${V4L2_LOOPBACK_VERSION}" ] || [ -L "/usr/src/${V4L2_LOOPBACK_DC}-${V4L2_LOOPBACK_VERSION}" ]; then
+    # Skipping link creation...
+    echo "'usr/src/${V4L2_LOOPBACK_DC}-${V4L2_LOOPBACK_VERSION}' already exists. Skipping creation."
+else
+    ln -s "`pwd`" "/usr/src/${V4L2_LOOPBACK_DC}-${V4L2_LOOPBACK_VERSION}"
+fi
+
+cd "/usr/src/${V4L2_LOOPBACK_DC}-${V4L2_LOOPBACK_VERSION}"
+dkms build -m ${V4L2_LOOPBACK_DC} -v ${V4L2_LOOPBACK_VERSION}
+dkms install -m ${V4L2_LOOPBACK_DC} -v ${V4L2_LOOPBACK_VERSION}
+dkms mkdeb -m ${V4L2_LOOPBACK_DC} -v ${V4L2_LOOPBACK_VERSION}
+
+create_v4l2loopback_dc_conf
```

Comment:
probably need `modprobe videodev` before this as well

Reply:
Nope you don't need to. This is exactly the reason why `modprobe` exists compared to `insmod`.
As long as your dependencies are correctly declared:
```
$ modinfo v4l2loopback-dc                  
filename:       /lib/modules/4.15.0-96-generic/updates/dkms/v4l2loopback-dc.ko
license:        GPL
srcversion:     02F912CBEDCDAE6A1C2C90C
depends:        videodev
retpoline:      Y
name:           v4l2loopback_dc
vermagic:       4.15.0-96-generic SMP mod_unload
parm:           width:frame width (int)
parm:           height:frame height (int)
```
`modprobe` will then load modules taking in account dependencies.

Here the demonstration:
```
$ lsmod |grep v4l
$ sudo modprobe v4l2loopback_dc
$ lsmod |grep v4l
v4l2loopback_dc        24576  0
videodev              184320  1 v4l2loopback_dc
```